tasks/verify-client-api-changes.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Compile extensions in order to check whether API changes
+ * are not breaking existing extensions.
+ */
+
+const execSync = require('child_process').execSync;
+const fs = require('fs');
+
+function getDirectories(path) {
+    return fs.readdirSync(path).filter((file) => {
+        return fs.statSync(path + '/' + file).isDirectory();
+    });
+}
+
+const allExtensions = 'scripts/extensions';
+
+getDirectories(allExtensions).forEach((extensionDir) => {
+    execSync(
```

Comment:
this would only print an error, but won't make the process fail

Reply:
did it? I had to add [this](https://github.com/superdesk/superdesk-client-core/blob/develop/run-end-to-end-tests.js#L59) otherwise failing tests were green on github actions / travis.

Reply:
In your case it was because of promises. Promises swallow errors by default. At least they used to. I remember being annoyed by it often. It looks like it improved since promises first appeared. I don't know whether the spec was changed or browsers started logging in anyway.

Reply:
ok if you're sure it works that's fine